src/client.rs

```diff
@@ -28,9 +28,10 @@ impl Client {
     /// # use meilisearch_sdk::{client::*, indexes::*};
     /// #
     /// // create the client
-    /// let client = Client::new("http://localhost:7700", "masterKey");
+    /// # let api_key = Option::Some(String::from("masterKey"));
+    /// # let client = Client::new("http://localhost:7700", api_key);
     /// ```
-    pub fn new(host: impl Into<String>, api_key: impl Into<String>) -> Client {
+    pub fn new(host: impl Into<String>, api_key: impl Into<Option<String>>) -> Client {
         Client {
             host: Rc::new(host.into()),
             api_key: Rc::new(api_key.into()),
```

Comment:
I don’t think there is anything in rust that implements `Into<Option<String>>`. And we can’t use it to do something like this;
```rust
Client::new("Hello", Some("World!"));
```

While accepting `Option<impl Into<String>>` would instead allow a lot of things.

```suggestion
    pub fn new(host: impl Into<String>, api_key: Option<impl Into<String>>) -> Client {
        Client {
            host: Rc::new(host.into()),
            api_key: Rc::new(api_key.map(|key| key.into()),
```

Reply:
I made the changes, I have 9 failing tests. Here is the output:
```
failures:

---- client::tests::test_error_delete_key stdout ----
thread 'client::tests::test_error_delete_key' panicked at 'assertion failed: matches!(error, Error ::\n    Meilisearch(MeilisearchError\n    {\n        error_code : ErrorCode :: ApiKeyNotFound, error_type : ErrorType ::\n        InvalidRequest, ..\n    }))', src/client.rs:709:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- client::tests::test_error_update_key stdout ----
thread 'client::tests::test_error_update_key' panicked at 'called `Result::unwrap()` on an `Err` value: Meilisearch(MeilisearchError { error_message: "The provided API key is invalid.", error_code: InvalidApiKey, error_type: Auth, error_link: "https://docs.meilisearch.com/errors#invalid_api_key" })', src/client.rs:871:48

---- client::tests::test_get_keys stdout ----
thread 'client::tests::test_get_keys' panicked at 'called `Result::unwrap()` on an `Err` value: Meilisearch(MeilisearchError { error_message: "The provided API key is invalid.", error_code: InvalidApiKey, error_type: Auth, error_link: "https://docs.meilisearch.com/errors#invalid_api_key" })', src/client.rs:685:44

---- client::tests::test_create_key stdout ----
thread 'client::tests::test_create_key' panicked at 'called `Result::unwrap()` on an `Err` value: Meilisearch(MeilisearchError { error_message: "The provided API key is invalid.", error_code: InvalidApiKey, error_type: Auth, error_link: "https://docs.meilisearch.com/errors#invalid_api_key" })', src/client.rs:758:48

---- client::tests::test_delete_key stdout ----
thread 'client::tests::test_delete_key' panicked at 'called `Result::unwrap()` on an `Err` value: Meilisearch(MeilisearchError { error_message: "The provided API key is invalid.", error_code: InvalidApiKey, error_type: Auth, error_link: "https://docs.meilisearch.com/errors#invalid_api_key" })', src/client.rs:698:48

---- client::tests::test_error_create_key stdout ----
thread 'client::tests::test_error_create_key' panicked at 'called `Result::unwrap()` on an `Err` value: Meilisearch(MeilisearchError { error_message: "The provided API key is invalid.", error_code: InvalidApiKey, error_type: Auth, error_link: "https://docs.meilisearch.com/errors#invalid_api_key" })', src/client.rs:805:66

---- client::tests::test_update_key stdout ----
thread 'client::tests::test_update_key' panicked at 'called `Result::unwrap()` on an `Err` value: Meilisearch(MeilisearchError { error_message: "The provided API key is invalid.", error_code: InvalidApiKey, error_type: Auth, error_link: "https://docs.meilisearch.com/errors#invalid_api_key" })', src/client.rs:835:52

---- search::tests::test_generate_tenant_token_from_client stdout ----
thread 'search::tests::test_generate_tenant_token_from_client' panicked at 'called `Result::unwrap()` on an `Err` value: Meilisearch(MeilisearchError { error_message: "The provided API key is invalid.", error_code: InvalidApiKey, error_type: Auth, error_link: "https://docs.meilisearch.com/errors#invalid_api_key" })', src/search.rs:591:14

---- tasks::test::test_wait_for_pending_updates_time_out stdout ----
thread 'tasks::test::test_wait_for_pending_updates_time_out' panicked at 'called `Result::unwrap_err()` on an `Ok` value: Succeeded { content: ProcessedTask { duration: 27.661999ms, enqueued_at: OffsetDateTime { utc_datetime: PrimitiveDateTime { date: Date { year: 2022, ordinal: 126 }, time: Time { hour: 12, minute: 28, second: 24, nanosecond: 508592000 } }, offset: UtcOffset { hours: 0, minutes: 0, seconds: 0 } }, started_at: OffsetDateTime { utc_datetime: PrimitiveDateTime { date: Date { year: 2022, ordinal: 126 }, time: Time { hour: 12, minute: 28, second: 24, nanosecond: 511721000 } }, offset: UtcOffset { hours: 0, minutes: 0, seconds: 0 } }, finished_at: OffsetDateTime { utc_datetime: PrimitiveDateTime { date: Date { year: 2022, ordinal: 126 }, time: Time { hour: 12, minute: 28, second: 24, nanosecond: 539383000 } }, offset: UtcOffset { hours: 0, minutes: 0, seconds: 0 } }, index_uid: "meilisearch_sdk-tasks-test-test_wait_for_pending_updates_time_out", update_type: DocumentAddition { details: Some(DocumentAddition { indexed_documents: Some(2), received_documents: 2 }) }, uid: 231 } }', src/tasks.rs:573:14


failures:
    client::tests::test_create_key
    client::tests::test_delete_key
    client::tests::test_error_create_key
    client::tests::test_error_delete_key
    client::tests::test_error_update_key
    client::tests::test_get_keys
    client::tests::test_update_key
    search::tests::test_generate_tenant_token_from_client
    tasks::test::test_wait_for_pending_updates_time_out

test result: FAILED. 34 passed; 9 failed; 0 ignored; 0 measured; 0 filtered out; finished in 5.03s
```
As far as the invalid API keys, would that be an issue of my running instance of Meilisearch? 
For the ```tasks::test::test_wait_for_pending_updates_time_out``` test, this was a new failure. I ran the tests before and it didn't fail. So I'm unsure of the behavior.